tez-api/src/main/java/org/apache/tez/client/TezClientUtils.java

```diff
@@ -336,8 +328,13 @@ public static FileSystem ensureStagingDirExists(Configuration conf,
     UserGroupInformation ugi = UserGroupInformation.getLoginUser();
     realUser = ugi.getShortUserName();
     currentUser = UserGroupInformation.getCurrentUser().getShortUserName();
-    if (fs.exists(stagingArea)) {
-      FileStatus fsStatus = fs.getFileStatus(stagingArea);
+    FileStatus fsStatus = null;
+    try {
+      fsStatus = fs.getFileStatus(stagingArea);
+    } catch (FileNotFoundException fnf) {
+      // Ignore
```

Comment:
what about returning if
```
if (fsStatus == null) {
  return fs;
}
```
and having the rest of the method unindented

Reply:
We can't return, there is an else block below if fsStatus is null
```
else {
      TezCommonUtils.mkDirForAM(fs, stagingArea);
    }
```